src/SSHDebugPS/PipeConnection.cs

```diff
@@ -196,17 +160,23 @@ public override List<Process> ListProcesses()
         /// Checks command exit code and if it is non-zero, it will throw a CommandFailedException with an error message if 'throwOnFailure' is true.
         /// </summary>
         /// <returns>true if command succeeded.</returns>
-        protected bool ExecuteCommand(string command, int timeout, bool throwOnFailure, out string commandOutput)
+        protected bool ExecuteCommand(string command, int timeout, bool throwOnFailure, out string commandOutput, out int exitCode)
         {
             commandOutput = string.Empty;
-            int exitCode = ExecuteCommand(command, timeout, out commandOutput);
+            exitCode = ExecuteCommand(command, timeout, out commandOutput);
             if (throwOnFailure && exitCode != 0)
             {
-                string error = String.Format(CultureInfo.InvariantCulture, StringResources.CommandFailedMessageFormat, command, exitCode, commandOutput);
+                string error = StringResources.CommandFailedMessageFormat.FormatCurrentCultureWithArgs(command, exitCode, commandOutput);
                 throw new CommandFailedException(error);
             }
             else
                 return exitCode == 0;
         }
+
+        protected bool ExecuteCommand(string command, int timeout, bool throwOnFailure, out string commandOutput)
+        {
+            int exitCode;
+            return ExecuteCommand(command, timeout, throwOnFailure, out commandOutput, out exitCode);
```

Comment:
>exitCode [](start = 91, length = 8)

Not sure if you're on C# 7, but if you are, you can use "_" as a built-in "discard" variable so you don't have to declare one just for the out parameter.

Reply:
Good to know. I don't know our entire build chain (Azure pipelines, devops, etc) but I do know we aren't supporting C#7.2 for named parameters.

---
In reply to: [297459596](https://github.com/microsoft/MIEngine/pull/884#discussion_r297459596) [](ancestors = 297459596)